### DISCUSSION TITLE
Not able to load local csv

### DISCUSSION THREAD

**Discussion by jeroenpullens** [grafana/grafana-csv-datasource#146] [Q&A] [answered]
Thank you very much for the package, but I am not able to load a local csv.

I get this error:
![image](https://user-images.githubusercontent.com/13275318/146187141-5e03bf97-8ca3-4fe4-bff3-1fbe018ec0ea.png)

What to do?

The file has a chmod 777 authorization

**Comment by dario-pilori** [comment]
Hi

I am also getting this error... very strange, because I tried and the grafana user can open (and edit it).

Did you manage to fix it?

**Reply by jeroenpullens** [reply]
No I haven't been able to fix it..

**Comment by ben-groves** [comment, marked as answer]
I think I found a fix...

1) I moved my folder containing `.csv` files out of my `/home/pi` folder (I am running on a pi too). I moved the folder to root : `/analytics`

2) I changed the owner of the folder and `.csv` files to `grafana` : `sudo chown grafana <folder or files>`

3) My folder `chmod` is `777`, the csvs are `666`.

My data sources now link without error.

**Comment by Alterdiag** [comment]
Hi, 

I have done the following steps and still have "stat /home/louis/Supervision/data_conso.csv: permission denied" error. 

1. Installing grafana
2. Installing marcusolsson-csv-datasource with this command line 
grafana-cli plugins install marcusolsson-csv-datasource
3. Add this at the begining of my grafana.ini file
[plugin.marcusolsson-csv-datasource]
allow_local_mode = true
4. Run the command (as suggested before) :
sudo chown grafana /home/louis/Supervision
chmod 777 /home/louis/Supervision
chmod 666 /home/louis/Supervision/data_conso.csv

It is still making the same error.

What could I try to get this resolved ? 

Thanks

**Reply by ben-groves** [reply]
Grafana has it's own system account `grafana`.
I think Linux is restricting access to the `louis` home folder.
Try moving `data_conso.csv` to `/Supervision/data_conso.csv`. 

Then
```
sudo chown grafana /Supervision ;
chmod 777 /Supervision ;
chmod 666 /Supervision/data_conso.csv ;
```
 Hope this helps.

**Reply by tabinashabir** [reply]
Anything for windows there?

**Comment by md-toro** [comment]
For Linux the below worked for me.  Still,  I think the documentation page needs an update.

Setup the data source as: /usr/share/grafana/public/
Optionally add a sub folder in the data source /usr/share/grafana/public/some_folder
When creating the panel either use the csv file in the data source or the relative path to the csv file
some_folder/sample.csv
